File: exercise/php/day08/tests/ToyProductionServiceTest.php
<?php

declare(strict_types = 1);

use Domain\Toy;
use PHPUnit\Framework\Assert;
use Services\ToyProductionService;
use Tests\Doubles\InMemoryToyRepository;

it('assigns toy to elf and sets state to IN_PRODUCTION', function (): void {
	$repository = new InMemoryToyRepository;
	$service = new ToyProductionService($repository);
	$toyName = 'Train';

	$repository->save(new Toy($toyName, Toy::UNASSIGNED));

	$service->assignToyToElf($toyName);

	$toy = $repository->findByName($toyName);
	Assert::assertEquals(Toy::IN_PRODUCTION, $toy->getState());
});

File: exercise/C#/day08/ToyProduction.Tests/ToyProductionServiceTest.cs
using FluentAssertions;
using ToyProduction.Domain;
using ToyProduction.Services;
using ToyProduction.Tests.Doubles;
using Xunit;

namespace ToyProduction.Tests;

public class ToyProductionServiceTest
{
    private const string ToyName = "Train";

    [Fact]
    public void AssignToyToElfShouldPassTheItemInProduction()
    {
        var repository = new InMemoryToyRepository();
        var service = new ToyProductionService(repository);
        repository.Save(new Toy(ToyName, State.Unassigned));

        service.AssignToyToElf(ToyName);

        repository.FindByName(ToyName)!
            .State
            .Should()
            .Be(State.InProduction);
    }
}